### ISSUE TITLE
Many aircraft show only lights, not the aircraft itself

### ISSUE THREAD

**Issue by SveWin53** [TwinFan/LiveTraffic#164]
2019-11-18 19:25:06 START

A.	ENVIRONMENT
01.	LiveTraffic v1.22.190829
02.	RealTraffic v8.2.77
03.	X-Plane v10.51.2
04.	X-Plane AI 20 planes defined using D:\SIM\X-Plane 10\Aircraft\XPFW\B737-800 SAS
05.	X-Plane airport used for test: D:\SIM\X-Plane 10\Custom Scenery\DAI Media Scenery - LEBL - Barcelona El Prat
05.	Windows 7 ULT 64-bit SP1

B.	ISSUE
01.	Planes are filling the buffer.
02.	Yellow labels are shown at the airport for all aircraft.
03.	Many aircraft show only lights, not the aircraft itself.
04.	After some time (more than 5 minutes), some aircraft start to appear with their labels.
05.	Some aircraft are never shown, only their banners.

C.	ATTACHED
01.	2019-11-18_145659 SNAP001.pdf - X-Plane 10 AI setup screen. Number of aircraft shown varies, probably with the number of active aircraft. Taken when not many aircraft were in the area.
02.	2019-11-18_161836 SNAP002.pdf - Showing structure and full content of CSL folder.
03.	CSL structure.txt - Showing structure and full content of CSL folder.
04.	LiveTraffic.prf - Active LiveTraffic settings.
05.	Log.txt - X-Plane 10 log with LiveTraffic DEBUG level entries.
06.	A320_1.png to A320_10.png - X-Plane snapshots with annotations.
NOTE:	AVI videos were created in X-Plane, but they were huge and with an unusable resolution, so deleted.
[2019-11-18 AC lights only shown.zip](https://github.com/TwinFan/LiveTraffic/files/3860285/2019-11-18.AC.lights.only.shown.zip)

**Comment by TwinFan** [comment]
This would actually be a support request better suited to the [Support Forum](https://forums.x-plane.org/index.php?/forums/topic/174691-if-you-want-support-contains-template/). Other than that: Perfectly documented!

The log doesn’t point to any actually error.

I am thinking, that all non-displayed aircraft are too low, sunken under the scenery and, once the center moved under the surface, they get cut off. My feeling is your scenery doesn’t align with the ground mesh. This is best visible in screenshot A320_10. The scenery looks level, but the 3 Aircraft are placed at 3 totally different altitudes as is perfectly visible by the location of the yellow labels. The non displayed one is way too low.

Can you please verify with standard scenery? Or at another standard scenery airport.

**Comment by TwinFan** [comment]
Or even simpler: take your own plane and roll left to right across those 3 parking positions shown in A320_10...it might be a roller coaster ride.

**Comment by SveWin53** [comment]
I am testing different airports now, must select the Cessna, the FF A320 takes forever to load...

I have taxied over those spots and most other places at that LEBL airport without noticing any dips in the terrain.  I disabled the payware LEBL airport as well as its accompanying mesh.  I saw the same behavior still.  I took snapshot of two aircraft taxiing behind each other, one was above ground and the other was invisible...  During taxiing the visible one stayed visible and the invisible one stayed invisible when rolling over the same track as the visible one...?

I am in the process of testing at EDDF, for which I do not have anything installed, just default X-Plane 10.

Will be back shortly...

-------------------------------------------------------------
Kind regards
     Svend Winsnes
-------------------------------------------------------------
| Private mail: svend.winsnes@gmail.com
| Snail mail....: Helsetdammen 43, NO-1353 Baerums Verk, NORWAY
| Phone..........: +47-6756-2757 (7/24)
| Mobile.........: +47-922-50-979 (7/24)
| Time Zone...: CEST (GMT+2)

**Comment by SveWin53** [comment]
I have tested at EDDF and again see invisible planes.  Attached are 4 .png snapshots with annotations:
*  A320_11 from LEBL default X-Plane 10 airport.
*  A320_12 from same.
* L5-1 (I changed aircraft to Stinson) from EDDF default X-Plane 10 airport.
* L5-2 from same.

I got to get some sleep now, will be back at this appx 1100Z tomorrow.
![A320_11](https://user-images.githubusercontent.com/57917307/69095803-6322a680-0a53-11ea-9cb0-dce25fb0e17d.png)
![A320_12](https://user-images.githubusercontent.com/57917307/69095805-6322a680-0a53-11ea-946e-d6f643c64e26.png)
![L5_1](https://user-images.githubusercontent.com/57917307/69095806-6322a680-0a53-11ea-9d34-d8c1ba00af15.png)
![L5_2](https://user-images.githubusercontent.com/57917307/69095808-6322a680-0a53-11ea-88a3-747faa53b98d.png)

**Comment by SveWin53** [comment]
I am fully prepared to be the culprit here :)  I just do not see it (yet)...

**Comment by SveWin53** [comment]
And NOTE:  Snapshot A320_11 id seen horizontally from ground level, the others are seen slightly from above. Can that account for difference in height between the labels?  (Final gasp before going to sleep :)

**Comment by TwinFan** [comment]
Weird...especially the cases where planes roll over the same spot...no idea at the moment...but will keep it in mind and come back in case of ideas.

**Comment by SveWin53** [comment]
Very good, thanks!  Wonderful product, I look forward to have it all :)

Many thanks for your support, it is much appreciated.

Kind regards Svend

**Comment by SveWin53** [comment]
Started at 1030Z today, and having a ball.... :)
2019-11-19 14:11:57 REPORT
A.	ACTIONS
01.	Disabled all plugins except LiveTraffic, in order to decrease log size and complexity.
02.	Disabled all of Custom Scenery (Airports, mesh, libraries, etc)
03.	Changed LiveTraffic AC labels to include: Transponder Hex Code, Altitude (ft), Speed, AC ICAO for screenshots.
04.	Turned on extra X-Plane on-screen Data (Upper left): location lat lon alt, Aircrafts (8 max) alt for screenshots.
05.	Turned on LiveTraffic debug logging.
06.	Started RealTraffic in LiveTraffic mode prior to starting X-Plane.
07.	Restarted X-Plane with Cessna 172SP.

B.	RESULTS
01.	X-Plane CTD shortly after startup completed and Cessna 172SP loaded, while RealTraffic is connecting.
02.	X-Plane does not crash if LiveTraffic plugin is disabled.

C.	ATTACHED
01.	2019-11-19_125536 SNAP001.pdf - Showing RealTraffic setup prior to loading X-Plane.
02.	Cessna_172SP_1.png - Snapshot showing X-Plane after load and prior to CTD.
02.	2019-11-19_130637 SNAP002.pdf - Showing System Event Log entry for crash.
03.	Log_1-20191119-1248.txt - X-Plane log.txt after CTD.

I need to create and execute a new action plan for this before we can continue with the subject issue....
In the meantime, given that you have some time and interest, if you get any ideas from this report please let me know?
Many thanks in advance, and kind regards
[2019-11-19 AC lights only shown.zip](https://github.com/TwinFan/LiveTraffic/files/3863964/2019-11-19.AC.lights.only.shown.zip)

**Comment by TwinFan** [comment]
Last line of log is
```
--=={This application has crashed because of the plugin: [XPLM_PLUGIN_XPLANE]}==--
```
A rare sight, when X-Plane claims to have caused the CTD itself.
You can search for a dump (see [here](https://twinfan.gitbook.io/livetraffic/reference/help-and-support#x-plane-quit-unexpectedly)) but it is highly unlikely that I can make anything of it, if it is really in XP code.

**Comment by SveWin53** [comment]
Hehe, I can't read those dumps either.  I was able to start X-Plane with LiveTraffic successfully when I did not have RealTraffic running.  EDDF is a very busy airport, with RealTraffic on (and X-Plane crashing) more than 50 aircraft were detected.  Maybe too many for X-Plane or LiveTraffic to swallow?

Anyway, as I was able to start X-Plane with LiveTraffic wo/ RealTraffic I now have a bunch of pictures, a video and an X-Plane log file.  I have processed the pictures and video and need to collect everything in a sensible manner.  Got to feed first, dinner time here.  Stay tuned for some maybe interesting 'stuff' later today...  Brgds

**Comment by SveWin53** [comment]
OK, I am back.  The log file I submitted earlier today was the wrong one.  It was due to too many files removed from the plugin folder.  The log you was meant to have has been overwritten.  And now, with RealTraffic connected there is no crash.  Also, only appx. 40 planes are seen at this time....

BAck to the original issue, I am building a package for you now.

**Comment by SveWin53** [comment]
2019-11-19 19:22:00 REPORT
A.	ACTIONS
01.	Started X-Plane w/ Cessna 172SP, OpenSky Live and wo/ RealTraffic connected.

B. RESULTS
01.	Lots of traffic in the area, both on ground and in the sky.
02.	Most of the AC are invisible, just LiveTraffic banner and AC lights.
03.	Even most of the inbound and landing traffic wo/ AC bodies, just lights.
04.	Ran this scenario for some time, capturing screenshots and some video.
05.	Very few of the 'Invisibles' morfed into Real aircraft. I was not able to catch these transformations on video as it always happened when I looked elsewhere (bad luck).  But the ones I saw that were stationary had the same altitude in their labels before and after becoming visible.
06.	This test was done with only the X-Plane default scenery and airport (EDDF). I noted that the ground seemed very uneven, also every aircraft had a different altitude in their banners.  This altitude shifted as they moved.  I tried to concentrate on stationary planes on the ground. I can do more of that if wanted.

C.	ATTACHED
01.	Cessna_172SP_2.png - Invisible planes in the sky, on final approach.
02.	Cessna_172SP_3.png - Visible and invisible planes on the ground. All on different altitudes.
03.	Cessna_172SP_4.png - Visible and invisible planes on the ground, almost same altitude (4ft diff).
04.	Cessna_172SP_5.png - More invisible planes in the sky.
05.	Cessna_172SP_6.png - Invisible plane landing.
06.	Cessna_172SP_7.png - Same plane, close to ground.
07.	Cessna_172SP_8.png - Yet another invisible plane on approach.
08.	Cessna_172SP_9.png - Same plane, on final.
09.	Cessna_172SP_10.png - Guys, stop filling those Follow Me cars with AvGas 130LL!!! I know it's fun, but...
10.	LiveTraffic samples 20191119-1702.mp4 - A few samples for study.
11.	Log_2-20191119-1458.txt - The X-Plane 10 log.txt with LiveTraffic debug entries.
All files in archive [2019-11-19 AC lights only shown (2).zip].

D. COMMENT
I am even more confused now, since airborne aircraft are also 'invisible' (not all but most). The BB CSL models seem to be picked up, otherwise there would be no lights shown (different lights according to model shown). But somehow, the real filled bodies are not shown, or sometimes materialize later even on stationery aircraft. Can there be something else in my system that causes this, I wonder?

I tested the first official version of LiveTraffic (where planes disappeared on the runway after landing) and invisible planes were not an issue then...

I really have no good ideas about this, except the testing shows how good your product has become! A fantastic idea and an impressive effort in making the planes behave so well (except for being invisible, and we do not know yet if this is caused by your product, I think).

I will stay on here for 2 hours more. But for the next 2 days I need to be in my studio recording a band. Friday is open, so far...
Best regards Svend
[2019-11-19 AC lights only shown (2).zip](https://github.com/TwinFan/LiveTraffic/files/3865590/2019-11-19.AC.lights.only.shown.2.zip)

**Comment by SveWin53** [comment]
'Good news'...  The band has postponed the studio session due to illness.  So, I will be here the next days.

**Comment by TwinFan** [comment]
Re Cessna_172SP_10.png...not too much coffee...too few information...[see FAQ](https://twinfan.gitbook.io/livetraffic/reference/faq#i-see-flying-follow-me-cars-whats-that). Not our issue.

I find it most astounding that when the plane is invisible that _all_ its lights are visible. There is a functionality to draw "lights only" for far away planes (which would be an object for analsis if something's broken there), but that would, if triggered, just paint artificial light blobs independend of the actual aircraft type. It would not paint strobe or nav lights.

So this is an issue with drawing the actual CSL object, which at its core is a standard X-Plane call. With thousands of parameters...so still LiveTraffic (or here: the libxplanemp, which handles drawing) can still do something wrong, but I have no clue what to look at.

So at the moment, until I have another idea, I'm afraid I can't help you.

**Comment by SveWin53** [comment]
Hi ,

The ...10.png was a joke :)  I am aware of the cause for this, I found that in the log.txt, about matching.  Not an issue that bothers me at all.  It was just funny, while I was concentrating on all those planes and their labels, to suddenly see that soaring car.  Made my day :))

To the actual issue:  I understand the complexity of this, and my issue report was based on a hope that this was either a known issue or an easy one to find and fix.  I understand fully that you can not spend a lot of time doing PD on one issue, especially if it is not widely seen among your clients.  If you do get an idea later, even if it is just some way for me to better do the PD, please remember me, will you?

Many thanks, anyway, for your time and efforts!  It was very much appreciated and it is very good to see a real support effort behind your product.  I will continue to investigate this myself, as time permits, and if unsuccessful I will look at the other very few solutions there are for X-Plane 10.

Have a very nice day, and thanks again for your efforts!

Kind regards Svend

**Comment by TwinFan** [comment]
Closed due to inactivity.
